common/document_parser/lib/section_parse/parsers/navy_parser.py

```diff
@@ -0,0 +1,165 @@
+from re import search, IGNORECASE, VERBOSE, compile, finditer, RegexFlag
+from os.path import splitext
+from typing import List
+from .parser_definition import ParserDefinition
+from .utils import (
+    match_number_hyphenated_section,
+    get_letter_dot_section,
+    match_number_dot_section,
+    match_first_appendix_title,
+    remove_pagebreaks,
+    DD_MONTHNAME_YYYY,
+)
+
+
+class NavyParser(ParserDefinition):
+    """Section parser for Navy document types (see SUPPORTED_DOC_TYPES).
+
+    Child of ParserDefinition.
+    """
+
+    SUPPORTED_DOC_TYPES = [
+        "opnavinst",
+        "opnavnote",
+        "bumedinst",
+        "milpersman",
+        "respersman",
+        "bupersinst",
+        "secnavinst",
+        "secnavnote",
+        "mco",
+        "navmc",
+        "comnavresforcominst",
+        "comnavresforcomnote",
+        "bumednote",
+        "alnav",
+        "mcrp",
+        "mcbul",
+        "fmfrp",
+    ]
+
+    def __init__(self, doc_dict: dict, test_mode: bool = False):
+        super().__init__(doc_dict, test_mode)
+        self._text = self.get_raw_text()
+        self._filename_without_extension = splitext(self._filename)[0]
+
+    @property
+    def purpose(self):
+        purposes = self._get_numbered_section_with_name("purpose", False)
+        # In some document types, a "Situation" section is used in place of a
+        # "Purpose" section.
+        if not purposes and self._doc_type.lower() == "mcbul":
+            # In some MCBUL documents, the purpose section is nested within
+            # the situation section.
+            situation = self._get_numbered_section_with_name("situation")
+            if situation:
+                purposes = [get_letter_dot_section(situation[0], "purpose")]
+
+        return purposes
```

Comment:
it seems like it is just MCBUL docs, but if they are lacking purpose/situation sections a list with an empty string is returned for the `purpose_section`
![Screen Shot 2022-11-18 at 11 28 34 AM](https://user-images.githubusercontent.com/37678069/202754122-499c12b1-e3b0-4b39-b820-001b9fd4d878.png)

Reply:
@mishoe Ok, I'll leave this comment unresolved for when we revisit Navy docs